common/src/main/java/io/netty/util/internal/NettyBlockHoundIntegration.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.util.internal;
+
+import io.netty.util.concurrent.FastThreadLocalThread;
+import reactor.blockhound.BlockHound;
+import reactor.blockhound.integration.BlockHoundIntegration;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * This class integrates Netty with BlockHound.
+ *
+ * It is public but only because of the ServiceLoader's limitations
+ * and SHOULD NOT be considered a public API.
+ */
+@UnstableApi
+public class NettyBlockHoundIntegration implements BlockHoundIntegration {
```

Comment:
You can you a trick to hide this class but still have it public:

```java 
final class Hidden {
  public static final class NettyBlockHoundIntegration implements BlockHoundIntegration {
    //...
  }
```

Then, in the service loader:

```
io.netty.util.internal.Hidden$NettyBlockHoundIntegration
```

Service loaders reflect on the inner class, but not the container, so normal java code can't reference it.

Reply:
TIL! Nice trick! :)

Reply:
@carl-mastrangelo TIL... thanks for the tip :)